### ISSUE TITLE
[trainer] Add a loss recorder option

### ISSUE THREAD

**Issue by frgfm** [frgfm/Holocron#241]
Once a model has been trained, it would be nice to analyze its main failures. For that, one could:
- go through the training set computing the loss (unreduced on the batch dimension)
- keep the N samples with the highest loss
- plot the image, the loss, the prediction and the target

A later option would be to display the CAM to better understand the model behaviour.
